File: vlife-admin-react/src/api/FormField.ts
import { Schema } from "@formily/react";
import apiClient from "@src/api/base/apiClient";
import { DbEntity, PageQuery, Result } from "@src/api/base";
import { FormEventVo } from "./FormEvent";
import { FormReactionVo } from "./FormReaction";
import { DataType } from '@src/dsl/base';
import { PageComponentPropDto } from '@src/api/PageComponentProp';
import { Form } from './Form';

export type javaNumberType="number"|"long"|"double"|"int"|"integer"|"short"|"float";

export type javaType="string"|"boolean"|"date"|"number"|"long"|"double"|"int"|"integer"|"short"|"float";

// 列表字段
export interface FormField extends DbEntity {
  title: string; // 客户端设置标题
  javaTitle: string; //java注释标题
  dataType:DataType ; // 数据类型
  fieldType:"string"|"boolean"|"date"|"number"; // 字段类别
  javaType:javaType;//java类型
  formId: string; // 所属表单
  formGroupCode?: string; //分组code
  formTabCode?: string; //页签code
  pathName: string; // 全路径
  fieldName: string; // 字段名
  entityType: string; //字段实体模型名称
  entityFieldName: string; //数据库模型字段
  // componentType: string; //组件类型（基础组件，业务组件）
  description: string; // 描述
  sort: number; // 顺序
  required: boolean; // 必填
  initialValues: string; // 默认值
  x_component: string; // 采用组件
  dictCode: string; // 字典的coden
  dataIndex: string; // formliy里的字段名称，默认一致
  x_decorator: string; // 包裹组件
  x_hidden: boolean; // 隐藏
  disabled: boolean; // 禁用(待删除)
  readOnly: boolean; // 只读(待删除)
  x_read_pretty:boolean;// 只读 
  x_decorator_props$gridSpan: number; //包裹元素所占grid列数
  x_component_props$placeholder: string; //input填写说明
  x_decorator_props$layout:  "vertical" | "horizontal"| undefined; //标签位置
  x_decorator_props$labelAlign: string; //标签对齐方式
  x_validator: string; //校验方式
  minLength:number;//最小长度
  maxLength:number;//最大长度
  minimum:number;//最小长度
  maximum:number;//最小长度
  vlife_pattern: string; //正则校验
  vlife_message: string; //校验不通过提醒
  componentSettingJson: string; //组件设置的JSON信息（可移除）
  hideLabel:boolean;//隐藏标签
  validate_unique:boolean;//表内唯一
  create_hide:boolean;//新增时隐藏
  modify_read:boolean;//修改时只读
  divider:boolean;//该列之前加入分割线
  dividerLabel:string;//分割线标题
  listHide: boolean; //列表展示
  listWidth:number;//列宽
  money:boolean;//是否金额
  listSort:number;//列排序
  listAlign:string; //对其方式 left|center|right
  safeStr:boolean;//安全展示的字符串（张**）
  listSearch:boolean; //列表的搜索条件模糊搜索
}

// 字段dto
export interface FormFieldDto extends FormField {
  //字段对应组件属性设置信息集合
  pageComponentPropDtos?: Partial<PageComponentPropDto>[];
}

/**
 * 字段模型属性
 */
 export interface FormFieldVo extends FormField, Schema {
  [key: string]: any; // 实现可以赋值任意字段
  form:Form,//所在模型信息
  title: string; // 标题
  //schema里包含字段
  required: boolean; // 必填
  readOnly: boolean; // 只读
  description: string; // 描述
  minLength:number;//最小长度
  maxLength:number;//最大长度
  minimum:number;//最小长度
  maximum:number;//最小长度
  form_type:string;//归属模型标识
  //--- vo注入字段
  events: FormEventVo[]; //当前field触发的事件(后端还没有)
  reactions: FormReactionVo[]; //字段的查询条件
  pageComponentPropDtos?: Partial<PageComponentPropDto>[];
  listWidth:number;//列宽
  money:boolean;//是否金额
  listSort:number;//列排序
  listHide:boolean
  listAlign:"left"|"center"|"right"; //对其方式 left|center|right
  listSearch:boolean; //列表的搜索条件模糊搜索
  safeStr:boolean;//安全展示的字符串（张**）

}

//字段详情
export const detail = (id: string): Promise<Result<FormField>> => {
  return apiClient.get(`/formField/detail/${id}`);
};

/**
 * 模型字段查询
 */
export const listAll = (params?: PageQuery): Promise<Result<FormField[]>> => {
  return apiClient.post(`/formField/list/all`,params||{});
};



/////////////////---------------------待删除
//可分组字段查询
export const listGroupOption = (params: {
  formId: string;
}): Promise<Result<FormField[]>> => {
  return apiClient.get(`/formField/list/groupOption`,{params});
};

/**
  * 主表字段查询
  * 外键字段来源表的所有字段信息
  */
 export const listRelationField = (params: {
  realationFieldId: string;
}): Promise<Result<FormField[]>> => {

  return apiClient.get(`/formField/list/relationField${params.realationFieldId?"?realationFieldId="+params.realationFieldId:""}`);
};

File: vlife/vlife-base/src/main/java/cn/wwwlike/form/entity/FormField.java
package cn.wwwlike.form.entity;

import cn.wwwlike.vlife.annotation.VClazz;
import cn.wwwlike.vlife.annotation.VField;
import cn.wwwlike.vlife.bean.DbEntity;
import lombok.Data;

import javax.persistence.Entity;
import javax.persistence.Table;

/**
 * 模型字段信息
 * 也包含UI场景的配置信息
 */
@Entity
@Data
@Table(name = "form_field")
@VClazz(orders = "sort_asc",remove = {FormReaction.class})
public class FormField extends DbEntity {
    /**
     * 标题
     * db标题
     */
    public String title;
    /**
     * 标题
     * java标题
     */
    public String javaTitle;

    /**
     * 字段名
     */
    public String fieldName;
    /**
     * 前端数据类型
     * basic,array,object
     */
    public String dataType;
    /**
     * 字段类型
     */
    public String fieldType;
    /**
     * 所属表单
     */
    public String formId;
    /**
     *  唯一不重复
     * validate是内置远程校验规则，触发form内部远程数据校验
     */
    public boolean  validate_unique;
    /**
     * 新增时隐藏
     */
    public boolean  create_hide;
    /**
     * 修改时只读
     * 不可修改
     */
    public boolean  modify_read;
    /**
     * 分组
     */
    public boolean divider;
    /**
     * 分组名称
     */
    public String dividerLabel;
    /**
     * 所在分组
     * 不存id原因，设计器阶段还没有产生id
     */
    public String formGroupCode;
    /**
     * 所在页签编码
     */
    public String formTabCode;
    /**
     * 标签隐藏
     */
    public boolean hideLabel;
    /**
     * 产生来源所在实体名称
     */
    public String entityType;
    /**
     * 字段来源实体对应字段名
     */
    public String entityFieldName;

    //组件类型
    public String componentType;

    /**
     * 全路径
     */
    public String pathName;
    /**
     * 字典的code
     */
    public String dictCode;

    //-----------formily 场景字段配置的信息

    /**
     * formliy里的字段名称，默认一致
     */
    public String dataIndex;

    /**
     * 默认值
     */
    public String initialValues;
    /**
     * 必填
     */
    public boolean required;
    /**
     * 隐藏
     */
    public boolean x_hidden;
    /**
     * 描述
     */
    public String description;
    /**
     * 采用组件
     */
    public String x_component;
    /**
     * 只读
     *
     */
    public Boolean x_read_pretty;
    /**
     * 禁用
     */
    public Boolean disabled;
    /**
     * 只读
     */
    public Boolean readOnly;
    /**
     * 顺序
     */
    public Integer sort;
    //-------------外观
    /**
     * 包裹组件
     */
    public String x_decorator;

    /**
     * vlife开头 需要特殊处理的字段
     */
    public String vlife_pattern;

    public String vlife_message;

    /**
     * 对应 x-decorator-props.gridSpan属性
     */
    public Integer x_decorator_props$gridSpan;

    /**
     * label位置 水平 vertical
     */
    public String x_decorator_props$layout;

    /**
     * label位置 对齐 left
     */
    public String x_decorator_props$labelAlign;
    /**
     * 描述信息 对应 x-component-props.placeholder;
     */
    public String x_component_props$placeholder;
    /**
     * 加载组件数据的地址关键字
     */
    public String apiKey;
    /**
     * 字段校验方式
     */
    public String x_validator;
    /**
     * 最小长度
     */
    public Integer minLength;
    /**
     * 最小长度
     */
    public Integer maxLength;
    /**
     * 最小值
     */
    public Double minimum;
    /**
     * 最大值
     */
    public  Double maximum;

    /**
     * 组件设置的JSON信息
     * 没有用到
     */
    public String componentSettingJson;

    // -------------------- 列表方面的设置 ---------------------
    /**
     * 列排序
     */
    public Integer listSort;
    /**
     * 列隐藏
     */
    public Boolean listHide;
    /**
     * 列宽
     */
    public Integer listWidth;
    /**
     * 是否金额
     * 格式化，数值型使用
     */
    public Boolean money;
    /**
     * 对齐方式
     * center|left|right
     */
    @VField(dictCode = "COLUMN_ALGIN")
    public String listAlign;

    /**
     * 列搜索
     * 放入统配搜若条件里
     */
    public boolean listSearch;

    /**
     * 数据库字段长度
     */
    public int dbLength;
    /**
     * 字符加密
     */
    public boolean safeStr;

    /**
     * 导入导出
     */
    public Boolean excel;
}
